src/types/composer.ts

```diff
@@ -343,80 +453,53 @@ export default class AlgokitComposer {
       if (isAbiValue(arg)) {
         methodArgs.push(arg)
-        return
+        continue
       }
 
-      if (Object.values(algosdk.ABITransactionType).includes(params.method.args[i + argOffset].type as algosdk.ABITransactionType)) {
-        let txn: algosdk.Transaction
-        switch (arg.type) {
-          case 'methodCall': {
-            const tempTxnWithSigners = this.buildMethodCall(arg, suggestedParams)
-            methodArgs.push(...tempTxnWithSigners)
-            argOffset += tempTxnWithSigners.length - 1
-            return
-          }
-          case 'appCall':
-            txn = this.buildAppCall(arg, suggestedParams)
-            break
-          case 'pay':
-            txn = this.buildPayment(arg, suggestedParams)
-            break
-          case 'assetOptIn':
-            txn = this.buildAssetTransfer({ ...arg, receiver: arg.sender, amount: 0n }, suggestedParams)
-            break
-          case 'assetCreate':
-            txn = this.buildAssetCreate(arg, suggestedParams)
-            break
-          case 'assetConfig':
-            txn = this.buildAssetConfig(arg, suggestedParams)
-            break
-          case 'assetDestroy':
-            txn = this.buildAssetDestroy(arg, suggestedParams)
-            break
-          case 'assetFreeze':
-            txn = this.buildAssetFreeze(arg, suggestedParams)
-            break
-          case 'assetTransfer':
-            txn = this.buildAssetTransfer(arg, suggestedParams)
-            break
-          case 'keyReg':
-            txn = this.buildKeyReg(arg, suggestedParams)
-            break
-          default:
-            throw Error(`Unsupported method arg transaction type: ${arg.type}`)
-        }
-
-        methodArgs.push({
-          txn,
-          signer: params.signer ? ('signer' in params.signer ? params.signer.signer : params.signer) : this.getSigner(params.sender),
-        })
-        return
+      if (isTransactionWithSigner(arg)) {
+        methodArgs.push(arg)
+        continue
       }
 
-      throw Error(`Unsupported method arg: ${arg}`)
-    })
+      if ('method' in arg) {
+        const tempTxnWithSigners = await this.buildMethodCall(arg, suggestedParams)
+        methodArgs.push(...tempTxnWithSigners)
+        continue
+      }
+
+      const txn = await arg
+      methodArgs.push({
+        txn,
+        signer: params.signer
+          ? 'signer' in params.signer
+            ? params.signer.signer
+            : params.signer
+          : this.getSigner(encodeAddress(txn.from.publicKey)),
+      })
+    }
 
     const methodAtc = new algosdk.AtomicTransactionComposer()
 
     methodAtc.addMethodCall({
       ...params,
       appID: Number(params.appId || 0),
-      note: params.note ? new Uint8Array(Buffer.from(params.note)) : undefined,
-      lease: params.lease ? new Uint8Array(Buffer.from(params.lease)) : undefined,
+      note: encodeTransactionNote(params.note),
+      lease: encodeLease(params.lease),
+      rekeyTo: params.rekeyTo,
```

Comment:
I realised we don't have fee control here - is that a problem?

Reply:
Added in [c16635b](https://github.com/algorandfoundation/algokit-utils-ts/pull/261/commits/c16635b32ae760344a7800ac81461f395821a543)